### DISCUSSION TITLE
How to create pdf from xml

### DISCUSSION THREAD

**Discussion by igivucko** [pymupdf/PyMuPDF#2461] [Q&A]
What would be the best way to insert data from the xml to the new pdf file?

Up until now I parsed the xml file with xmltodict library. Then flattened the nested dictionary, iterated through it and used insert_text function. This works but I am wondering is there some more appropriate method.

I also tried 
doc = fitz.open(xml)
  xml = doc.convert_to_pdf()
  pdf = fitz.open("pdf", xml)
  for page in pdf:
      text = page.get_text()   

this would work but xml file is in shift-JIS encoding and I don't get the correct output.
Does anyone have any suggestions

**Comment by JorjMcKie** [comment]
Text extraction works for all document types. That happens if you extract the text of the XML page?

**Reply by igivucko** [reply]
Yes, the xml is in the shift-js encoding. 
When I convert xml to dict I  use shift js encoding. I add japanese font in pymupdf and insert_text works fine.
But when I use the other method, I am able to open xml, convert it but when I use get_text method the text is not in the proper encoding.
